staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -230,7 +232,7 @@ func (pf *PortForwarder) forward() error {
 	select {
 	case <-pf.stopChan:
 	case <-pf.streamConn.CloseChan():
-		runtime.HandleError(errors.New("lost connection to pod"))
+		return ErrLostConnectionToPod
```

Comment:
I think making it clear to callers we hit this path could make sense.

would we expect `kubectl port-forward` to exit with a non-zero return code in this return path?

Reply:
Yes, I think whatever the general error exit code is for the platform (ie. 1 for Linux) would be appropriate here... It was running and something unexpected happened, resulting in the executable terminating.

Technically, I think Ctrl+C should result in exit code 130 (at least on Linux), but it will continue to return 0, as I haven't made any direct changes to how kubectl decides which return code to exit with.

---
Here are a couple of scenarios with the code from this PR:

Intentional exit using Ctrl+C, exit code remains 0:
```
~/go/src/k8s.io/kubernetes (pf-exit) $ _output/bin/kubectl port-forward acid-minimal-cluster-0 5432
Forwarding from 127.0.0.1:5432 -> 5432
Forwarding from [::1]:5432 -> 5432
^C~/go/src/k8s.io/kubernetes (pf-exit) $ echo $?
0
```

Unintentional exit due to pod connection lost, exit code will be 1 now:
```
~/go/src/k8s.io/kubernetes (pf-exit) $ _output/bin/kubectl port-forward acid-minimal-cluster-0 5432
Forwarding from 127.0.0.1:5432 -> 5432
Forwarding from [::1]:5432 -> 5432
Handling connection for 5432
Handling connection for 5432
E1223 09:04:25.833912   47643 portforward.go:409] an error occurred forwarding 5432 -> 5432: error forwarding port 5432 to pod 955b967f6206b8037a7587a2701fb6a8b81f919265488e6fc8153c1d1a1be56e, uid : failed to execute portforward in network namespace "/var/run/netns/cni-fc4a08ea-2ea2-d564-7940-e7e059a207a8": read tcp4 127.0.0.1:40372->127.0.0.1:5432: read: connection reset by peer
error: lost connection to pod
~/go/src/k8s.io/kubernetes (pf-exit) $ echo $?
1
```